src/util.ts

```diff
@@ -1,14 +1,17 @@
-import { AxiosError } from 'axios';
+import axios, { AxiosError } from 'axios';
 
 export function serializeObject(obj: unknown): object {
   if (obj && typeof obj === 'object') {
     return Object.getOwnPropertyNames(obj).reduce((map, key) => {
+      const propertyValue = [key];
       // eslint-disable-next-line no-param-reassign
-      map[key] = obj[key];
+      map[key] =
+        key === 'error' && axios.isAxiosError(propertyValue)
```

Comment:
Wouldn't this fit better into `serializeAxiosError` below?

Reply:
Or in the Exception where the `error` object is being passed. I think the confusion comes from this being a "generic" serialize method and an ambiguous `error` property.

Reply:
Or even, we could use `serializeAxiosError` in the client exception

Reply:
@bia95 brought the point that the axios error might end up as details in any exception, not only the client exception. So it might be worth using axios.isAxiosError in the base  `Exception` class using the `serializeAxiosError` as you propose.

Reply:
Well anything can end up anywhere in theory right. Doesn't even need to be Exception - we can return object contaning axios error in the non-error response handler too. Doesn't even need to be axios error. Doesn't even need to contain JWT, can be different sensitive data.

What I'm trying to say - it's easier to fix the real cases. The space of hypothetical cases is limitless and a lot harder to code against. AFAIK we only pass AxiosErrors to ClientException in the HTTP client. We can only do it differently if we don't use the HTTP client, but axios directly. Which we don't do.

Reply:
I think it makes sense to be here since this function takes care of the details/errors. I'd remove the `key === 'error'`, that it's not needed and for every Axios error found it would only use the response data. 
I would not limit the change to 'ClientException`, this can happen to other exception classes as well. I see serializeAxionsError is used when the logger so it just hides the other error data and I don't see why do we want to keep the header data around.

Reply:
o ..wow .. I didn't refresh my page ..now I see your responses

Reply:
Ok but this way we will have 3 places where we handle AxiosError differently in lambda-essentials-ts. Perhaps we could do it in one place and consistently instead?

Reply:
Oh I also didn't refresh page after you refreshed your page 🤦 Come on GitHub! :D

Reply:
ClientException

Reply:
My 2 cents:
- the HTTP client logs that use `serializeAxiosError` are not that useful without body/headers. I never managed to use them to understand what happened, always need to run a service locally and need to log the request body to find why the call is failing. That's said, `correlationId` might improve the situation if we are able to find request body's on the "server" side of the call
- we should really try to not have 3 places that are responsible for the same thing and each of them does it differently

Reply:
The `serializeAxiosError` shouuuuuld log the body https://github.com/Cimpress-MCP/lambda-essentials-ts/pull/8/files#diff-3294a832ea2276e554177e0b3007cc2d401c082912c7fbde49fa09141bf1aed1R26

But to explain better what I was suggesting above - what about when we catch `AxiosError` in HttClient we immediately convert it into a `ClientException` (and only pick the useful stuff there). Then we can log and return `ClientException` that will be safe and readable everywhere. And we won't need to consider AxiosError anywhere else because it won't be possible for them to get out of the HttpClient

Reply:
Talked with @akincel and in the end, I think we should just keep it simple. We should go with your initial suggestion @mmaruniak  and just call `serializeAxiosError` in ClientException.